portal/src/main/webapp/js/src/oncoprint/setup.js

```diff
@@ -2499,9 +2499,153 @@ window.CreateCBioPortalOncoprintWithToolbar = function (ctr_selector, toolbar_se
 		    cbio.download.initDownload(content, downloadOpts);
 		});
 	    });
+	    
+	    $('body').on('click', '.oncoprint-tabular-download', function () {
+		//Get caseNames (either patients or samples)
+		window.QuerySession.getUIDToCaseMap().then(function (uid_to_case) {
+		    var caseNames = [];
+		    oncoprint.getIdOrder().map(function (uid) {
+			caseNames.push(uid_to_case[uid]);
+		    });
+		    var prefixName = (State.using_sample_data ? "SAMPLE_DATA_" : "PATIENT_DATA_"); //Name depending on the type of case
+
+		    //Gather all the Oncoprint data
+		    var oncoprintData = {
+			    'CLINICAL': {},
+			    'CNA': {},
+			    'MUTATIONS': {},
+			    'MRNA': {},
+			    'PROTEIN': {},
+			    'FUSION': {},
+			    'GSVA': {}
+		    };
+		    
+		    //Create maps for genetic data
+		    var cnaMap = {
+			    'amp': 'Amplification',
+			    'gain': 'Gain',
+			    'hetloss': 'Shallow Deletion',
+			    'homdel': 'Deep Deletion'
+		    };
+		    var mutationMap = {};
+		    if (State.colorby_knowledge) {
+			mutationMap = {
+				    'inframe': 'Inframe Mutation (putative passenger)',
+				    'inframe_rec': 'Inframe Mutation (putative driver)',
+				    'missense': 'Missense Mutation (putative passenger)',
+				    'missense_rec': 'Missense Mutation (putative driver)',
+				    'promoter': 'Promoter Mutation',
+				    'promoter_rec': 'Promoter Mutation',
+				    'trunc': 'Truncating mutation (putative passenger)',
+				    'trunc_rec': 'Truncating mutation (putative driver)'
+			    };
+		    } else {
+			mutationMap = {
+				    'inframe': 'Inframe Mutation',
+				    'inframe_rec': 'Inframe Mutation',
+				    'missense': 'Missense Mutation',
+				    'missense_rec': 'Missense Mutation',
+				    'promoter': 'Promoter Mutation',
+				    'promoter_rec': 'Promoter Mutation',
+				    'trunc': 'Truncating mutation',
+				    'trunc_rec': 'Truncating mutation'
+			    };
+		    }
+		    var mrnaMap = {
+			    'up': 'mRNA Upregulation',
+			    'down': 'mRNA Downregulation'
+		    };
+		    var proteinMap = {
+			    'down': 'Protein Downregulation',
+			    'up': 'Protein Upregulation'
+		    };
+		    var fusionMap = {
+			    'true': 'Fusion'
+		    };
+
+		    //Add genetic data
+		    var track_ids = utils.objectValues(State.genetic_alteration_tracks);
+		    patientOrSample = (State.using_sample_data ? "sample" : "patient");
+		    for (var track_id=0; track_id<track_ids.length; track_id++){
+			var currentTrackData = oncoprint.getTrackData(track_ids[track_id]);
```

Comment:
a more principled way to do both of these is to use QuerySession.get(Sample/Patient)GenomicEventData - the oncoprint_data attribute of those elements is the track data, and gene is on there.

Reply:
I've tried to do it with `QuerySession.get(Sample/Patient)GenomicEventData`, but this data do not reflect if the mutations are putative drivers or passengers. `oncoprint.getTrackData` does reflect this.

Reply:
if you call it with the argument `true` it will reflect that

`QuerySession.get(..)GenomicEventData(true)`